### PR TITLE
feat: add PolicyStatus for EnvoyPatchPolicy

### LINKED ISSUE #2631: Replace Status with PolicyAncestorStatus in CTP, BTP and SP
*Description*:
>Describe the issue.

Use the upstream `PolicyAncestorStatus` definition which creates a top level structure per controller
https://gateway-api.sigs.k8s.io/geps/gep-713/#standard-status-struct

instead of what we have today e.g. https://github.com/envoyproxy/gateway/blob/1775624114aedee731bee565a0a100eeee80fc13/api/v1alpha1/backendtrafficpolicy_types.go#L97

Should fix https://github.com/envoyproxy/gateway/issues/2520

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2631

### CODE REVIEW

File: internal/cmd/egctl/status_test.go [hunk 1]
```diff
@@ -518,53 +516,7 @@ btls      foobar2   test-status-2   test reason 2
...
-			verbose:            false,
-			allNamespaces:      false,
-			typedName:          true,
-			outputs: `NAME                   TYPE      STATUS          REASON
-envoypatchpolicy/epp   foobar2   test-status-2   test reason 2
-                       foobar1   test-status-1   test reason 1
-`,
-			expect: true,
-		},
+		// TODO(sh2): add a policy status test for egctl x status cmd
```
Comment (shawnh2) [inline internal/cmd/egctl/status_test.go:519]: will update `egctl x status` as a follow-up

File: internal/xds/translator/testdata/out/xds-ir/jsonpatch-missing-resource.envoypatchpolicies.yaml [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
-      type: Programmed
+    ancestors:
+    - ancestorRef:
+        group: gateway.networking.k8s.io
+        kind: Gateway
+        name: foobar
+        namespace: default
+      conditions:
+      - lastTransitionTime: null
+        message: 'Unable to find xds resources: type.googleapis.com/envoy.config.listener.v3.Listener/non-existant-listener'
```
Comment (zhaohuabing) [inline internal/xds/translator/testdata/out/xds-ir/jsonpatch-missing-resource.envoypatchpolicies.yaml:12]: nit :
```suggestion
        message: 'Unable to find xds resources: type.googleapis.com/envoy.config.listener.v3.Listener/non-existing-listener'
```

**Review by zhaohuabing — COMMENTED** [review]
This looks really great! Just a tiny nit.

File: internal/xds/translator/testdata/out/xds-ir/jsonpatch-missing-resource.envoypatchpolicies.yaml (hunk 2 above)
Comment (shawnh2) [reply internal/xds/translator/testdata/out/xds-ir/jsonpatch-missing-resource.envoypatchpolicies.yaml]: fixed

**Review by zirain — APPROVED** [review]

**Review by Xunzhuo — APPROVED** [review]
Looks like all of our polcies has been implemented the policystatus, great job!

**Pull request merged**